cl/search/factories.py

```diff
@@ -209,20 +209,23 @@ class OpinionClusterWithParentsFactory(
     pass
 
 
-class RECAPDocumentFactory(DjangoModelFactory):
+class DocketEntryFactory(DjangoModelFactory):
     class Meta:
-        model = RECAPDocument
+        model = DocketEntry
 
     description = Faker("text", max_nb_chars=750)
-    document_type = RECAPDocument.PACER_DOCUMENT
-    pacer_doc_id = Faker("pyint", min_value=100_000, max_value=400_000)
+    docket = SubFactory("cl.search.factories.DocketFactory")
 
 
-class DocketEntryFactory(DjangoModelFactory):
+class RECAPDocumentFactory(DjangoModelFactory):
     class Meta:
-        model = DocketEntry
+        model = RECAPDocument
 
     description = Faker("text", max_nb_chars=750)
+    docket_entry = SubFactory(DocketEntryFactory)
+    document_type = RECAPDocument.PACER_DOCUMENT
+    pacer_doc_id = Faker("pyint", min_value=100_000, max_value=400_000)
+    is_available = Faker("boolean", chance_of_getting_true=40)
```

Comment:
I think this is why your test is failing. The `AbstractPacerDocument` class, `is_available` defaults to False. Before this change, that meant that all faked RECAPDocuments would have a false value. Now 40 of them are true, so some tests will fail.

If you need this value for your test, you can just tweak the other test to explicitly set it to False.

Reply:
Sorry, Mike I'm a bit confused by your comment. You're saying the original issue is the 

`AbstractPacerDocument` class defaulting to false, as seen here:

```python3
      is_available = models.BooleanField(
          help_text="True if the item is available in RECAP",
          blank=True,
          null=True,
          default=False,
      )
```

Therefore, the code in `test_available_only_field` should explicitly set `is_available` to False, but isn't that already what it's doing?

```python3
      recap_document = RECAPDocumentFactory(
          docket_entry=entry,
          description="New File",
          document_number="1",
          is_available=False,
          page_count=5,
      )
```

P.S.: could you show me how to reference lines of code inline, like your comment does?

Reply:
Hi @v-anne,  yeah Mike is right. The issue with the `RECAPDocumentFactory` and the failing tests is that we have instances of `RECAPDocumentFactory` across our codebase that don't explicitly set the `is_available` field. For example:

https://github.com/freelawproject/courtlistener/blob/8819ac1d1d5ce111769f4fceb7a46995bd606f7a/cl/lib/test_helpers.py#L1266

In those factories, we expect `is_available` to be `False`, as defined in the `AbstractPacerDocument` class.

However, the introduction of `is_available = Faker("boolean", chance_of_getting_true=40)` is causing these factories to occasionally assign a `True` value, which leads to failing tests.

I recommend removing `is_available` from the `RECAPDocumentFactory` class and explicitly setting the `is_available` values in the test as needed. For example, you can create a loop that instantiates `RECAPDocumentFactory`, assigning `is_available` randomly within the loop.


> P.S.: could you show me how to reference lines of code inline, like your comment does?

You can add a reference in the comment during a code review on the `Files Changed` tab by simply clicking the line of code and posting a comment. You can also select multiple lines at once.
![Screenshot 2024-09-13 at 1 49 32 p m](https://github.com/user-attachments/assets/980d5353-9b38-4453-a69f-4cd5ba64a6ed)

You can also reference code by using the permalink of a specific line, as I did in the comment above.
![Screenshot 2024-09-13 at 1 48 50 p m](https://github.com/user-attachments/assets/5b729473-b04e-4874-8fd6-f88fbe9db21b)